File: android/src/main/kotlin/tech/shmy/a_player/player/APlayerInterface.kt
package tech.shmy.a_player.player

import android.view.Surface

interface APlayerInterface {
    val duration: Long
    val speed: Float
    val isLoop: Boolean
    fun addListener(listener: APlayerListener)
    fun setSurface(surface: Surface)
    fun play()
    fun pause()
    fun stop()
    fun setHttpDataSource(url: String, startAtPositionMs: Long, headers: Map<String, String>)
    fun setFileDataSource(path: String, startAtPositionMs: Long)
    fun release()
    fun prepare()
    fun seekTo(positionMs: Long)
    fun setSpeed(speed: Float)
    fun setLoop(isLoop: Boolean)
}

File: ios/Classes/Player/APlayerInterface.swift
//
//  APlayerInterface.swift
//  a_player
//
//  Created by drm-chao wang on 2022/7/4.
//

import Foundation

protocol APlayerInterface {
    var duration: Int64 { get }
    var speed: Float { get }
    var isLoop: Bool { get }
    var isAutoPlay: Bool { get }
    func addListener(listener: APlayerListener) -> Void
    func play() -> Void
    func pause() -> Void
    func stop() -> Void
    func enableHardwareDecoder(enabled: Bool) -> Void
    func destroy() -> Void
    func prepare(isAutoPlay: Bool) -> Void
    func seekTo(positionMs: Int64) -> Void
    func setSpeed(speed: Float) -> Void
    func setLoop(isLoop: Bool) -> Void
}
